src/InfoBarWidget.js

```diff
@@ -0,0 +1,122 @@
+var InfoBarWidget = function InfoBarWidget( config ) {
+	var widget = this;
+
+	// Configuration initialization
+	config = config || {};
+
+	// Parent constructor
+	OO.ui.ButtonWidget.parent.call( this, config );
+	OO.ui.mixin.IconElement.call( this, config );
+	OO.ui.mixin.LabelElement.call( this, config );
+	OO.ui.mixin.TitledElement.call( this, config );
+	OO.ui.mixin.FlaggedElement.call( this, config );
+
+	this.closeIcon = new OO.ui.IconWidget( {
+		icon: 'clear',
+		flags: [ 'invert' ],
+		classes: [ 'ext-wwt-infoBarWidget-close' ]
+	} );
+	this.userInfoLabel = new OO.ui.LabelWidget( {
+		label: mw.msg( 'ext-whowrotethat-ready-general' ),
+		classes: [ 'ext-wwt-infoBarWidget-info' ]
+	} );
+	this.$pendingAnimation = $( '<div>' )
+		.addClass( 'ext-wwt-infoBarWidget-spinner' )
+		.append(
+			$( '<div>' )
+				.addClass( 'ext-wwt-infoBarWidget-spinner-bounce' )
+		);
+
+	// Set properties
+	this.setState( config.state || 'pending' );
+	this.setLabel( mw.msg( 'ext-whowrotethat-state-pending' ) );
+
+	// Scroll event
+	$( window ).scroll( () => {
+		var pos = $( '#bodyContent' )[ 0 ].getBoundingClientRect();
+
+		widget.$element.toggleClass(
+			'ext-wwt-infoBarWidget-sticky',
+			pos.top < widget.$element.height()
+		);
+	} );
+
+	// Close event
+	this.closeIcon.$element.on( 'click', function () {
+		widget.emit( 'close' );
+	} );
+
+	// Initialize
+	this.$element
+		.addClass( 'ext-wwt-infoBarWidget' )
+		.append(
+			this.$pendingAnimation,
+			this.$icon,
+			this.$label,
+			this.userInfoLabel.$element,
+			this.closeIcon.$element
+		);
+};
+
+/* Setup */
+
+OO.inheritClass( InfoBarWidget, OO.ui.Widget );
+OO.mixinClass( InfoBarWidget, OO.ui.mixin.IconElement );
+OO.mixinClass( InfoBarWidget, OO.ui.mixin.LabelElement );
+OO.mixinClass( InfoBarWidget, OO.ui.mixin.TitledElement );
+OO.mixinClass( InfoBarWidget, OO.ui.mixin.FlaggedElement );
+
+/**
+ * Define legal states for the widget
+ *
+ * @type {Array}
+ */
+InfoBarWidget.static.legalFlags = [ 'pending', 'ready', 'err' ];
+
+/**
+ * Change the state of the widget
+ *
+ * @param {string} state Widget state; 'pending', 'ready' or 'error'
+ */
+InfoBarWidget.prototype.setState = function ( state ) {
+	var flags = {};
+
+	if ( this.state !== state ) {
+		this.constructor.static.legalFlags.forEach( ( flag ) => {
```

Comment:
`( flag )` can just be `flag`, if it matters (maybe we should set some additional rules in eslint?)

Reply:
Aye, it probably can be; I was just really used to the parentheses. I'm actually not sure what our eslint rules are on that; I'll give it a try and see if eslint lets it in.

Reply:
I don't think we're using any for this, but we could https://eslint.org/docs/rules/arrow-parens. Some enforce the parentheses others enforce it not be there. Very trivial, I just love how there's a lot less syntax in ES6+!

Reply:
I added the rule to the `.eslintrc` file:

```
        "arrow-parens": [ "error", "as-needed" ]
```